third_party/CMakeLists.txt

```diff
@@ -29,3 +29,5 @@
 if(NOT OT_EXTERNAL_MBEDTLS)
     add_subdirectory(mbedtls)
 endif()
+
+add_subdirectory(tcplp)
```

Comment:
Do we always want to add the `tcplp` subdir? Or be conditional on a CMake option?

Reply:
I believe there is a config option for enabling TCP, but I’m not sure if it’s a CMake option that is understood by the Makefile. I’ll look into it.

Other than that flag, I can’t think of anything else we might check before adding this.

Reply:
Hi Jonathan,
I looked into this. I believe that the current OpenThread configuration flag for TCP (`OPENTHREAD_CONFIG_TCP_ENABLE`) can't be inspected in `CMakeLists.txt`; it's not in `options.cmake`.

There's the question if we should add a CMake option for this. I feel it would be confusing to add it, since it would have to be kept in sync with the `OPENTHREAD_CONFIG_TCP_ENABLE` configuration flag; it wouldn't make sense to enable one but not the other. Additionally, my understanding is that even without a CMake option, `OPENTHREAD_CONFIG_TCP_ENABLE` is `0`, then the TCPlp library won't be used, and the linker will optimize it out. I checked this by using `arm-none-eabi-objdump` to inspect the final executable files and verify that `OPENTHREAD_CONFIG_TCP_ENABLE` set to `0` implies that the TCPlp functions aren't included in the final executable.

So the only advantage I can think of for adding a CMake option is to speed up the build when it is disabled (since the TCPlp files won't have to be compiled when the CMake option is disabled. Personally, I don't see enough of an advantage to doing this to justify the complexity of an additional CMake option that needs to be kept in sync with the existing `OPENTHREAD_CONFIG_TCP_ENABLE` flag.

@jwhui What is your opinion on this?

Reply:
I'm fine with the current PR. Thanks.